http/src/response/mod.rs

```diff
@@ -242,9 +242,7 @@ impl<T> Response<T> {
 
             // Create a buffer filled with zeroes so we can copy the aggregate
             // body into it.
-            let mut buf = iter::repeat(0)
-                .take(aggregate.remaining())
-                .collect::<Vec<_>>();
+            let mut buf = vec![0; aggregate.remaining()];
```

Comment:
Couldn't `From<[T; N]>` be used here? Or would that cause a double allocation?

Reply:
That would require a constant length. `vec!` has an implementation that does quite a lot of unsafe stuff to achieve its performance, so alternatives like reserving and filling would be slower.

Reply:
>
>
> Normally we try to avoid new usages of macros, but because this is the only way the standard library exposes this functionality we'll have to take it. Please leave a note justifying the use of the macro with evidence (such as linking to the RFC).

I think `vec![]` should be exempt from that rule due to the reasons in that RFC & to the fact that it's one of the most popular macros. I can add a comment saying "`vec![]` is the most optimized way to create a buffer" (or whatever you want) but it feels a bit weird to justify.